CPDShell/Core/algorithms/kliep_cpd.py

```diff
@@ -0,0 +1,75 @@
+from collections.abc import Iterable
+
+import numpy as np
+
+from CPDShell.Core.algorithms.DensityBasedCPD.abstracts import DensityBasedAlgorithm
+
+
+class KliepAlgorithm(DensityBasedAlgorithm):
+    """Kullback-Leibler Importance Estimation Procedure (KLIEP) algorithm
+    for change point detection.
+
+    KLIEP estimates the density ratio between two distributions and uses
+    the importance weights for detecting changes in the data distribution.
+    """
+
+    def __init__(self, bandwidth, regularization_coef, threshold):
+        """Initialize the KLIEP algorithm.
+
+        Args:
+            bandwidth (float): bandwidth parameter for density estimation.
+            regularization_coef (float): regularization parameter.
+            threshold (float, optional): threshold for detecting change points.
+            Defaults to 1.1.
+        """
+        self.bandwidth = bandwidth
+        self.regularization_coef = regularization_coef
+        self.threshold = threshold
+
+    def detect(self, window: Iterable[float]) -> int:
+        """Detect the number of change points in the given data window
+        using KLIEP.
+
+        Args:
+            window (Iterable[float]): the data window to detect change points.
+
+        Returns:
+            int: the number of detected change points.
+        """
+        weights = self._calculate_weights(
+            window,
+            window,
+            self.bandwidth,
+            self.regularization_coef,
+            lambda w, alpha: -np.mean(w) + self.regularization_coef * np.sum(alpha**2),
+        )
+
+        change_points = 0
+        for time, weight in enumerate(weights):
```

Comment:
Also there's no need in `enumerate`, because here we don't use `time` as far as I see. We can use just `for weight in weights:`